test/upgrade/upgrade_adapter.spec.ts

```diff
@@ -22,46 +20,33 @@ class MockAngularUpgradeAdapter {
   }
 }
 
+let instantiatedNgUpgradeAdapter: MockAngularUpgradeAdapter
 let upgradeAdapter: NgMetadataUpgradeAdapter
 
 describe( `upgrade`, () => {
 
   describe( `NgMetadataUpgradeAdapter`, () => {
 
     let sandbox: Sinon.SinonSandbox;
-    beforeEach( () => {
-      sandbox = sinon.sandbox.create();
-    } );
-    afterEach( () => {
-      sandbox.restore();
-    } );
 
     beforeEach( () => {
-      upgradeAdapter = new NgMetadataUpgradeAdapter( MockAngularUpgradeAdapter );
-    } )
-
-    afterEach( () => {
-      sandbox.restore();
-    } );
 
-    it( `should set the internal _upgradeAdapter singleton`, () => {
-      expect( JSON.stringify( upgradeAdapter._upgradeAdapter ) ).to.equal( JSON.stringify( new MockAngularUpgradeAdapter() ) );
-    } );
-
-    describe( `#addProvider`, () => {
-
-      it( `should call the internal adapter's addProvider() method with the given provider`, () => {
-
-          sandbox.spy( upgradeAdapter._upgradeAdapter, 'addProvider' );
+      sandbox = sinon.sandbox.create();
 
-          class Ng2Provider {}
+      @NgModule({})
+      class AppModule {}
 
-          upgradeAdapter.addProvider( Ng2Provider );
+      instantiatedNgUpgradeAdapter = new MockAngularUpgradeAdapter(forwardRef(() => AppModule))
```

Comment:
yeah, in our case this is not needed because we are defining `AppModule` class before we create adapter, so pls either remove the `forwardRef` or add there some quick comment with explanation